### ISSUE TITLE
Overflow in imabalance fee calculation

### ISSUE THREAD

**Issue by palango** [raiden-network/raiden#5026]
```
channel_capacity = 100000, proportional_imbalance_fee = 1

    def calculate_imbalance_fees(
        channel_capacity: TokenAmount, proportional_imbalance_fee: ProportionalFeeAmount
    ) -> Optional[List[Tuple[TokenAmount, FeeAmount]]]:
        """ Calculates a U-shaped imbalance curve
    
        The penalty term takes the following value at the extrema:
        channel_capacity * (proportional_imbalance_fee / 1_000_000)
        """
        assert channel_capacity >= 0
        assert proportional_imbalance_fee >= 0
    
        if proportional_imbalance_fee == 0:
            return None
    
        if channel_capacity == 0:
            return None
    
        MAXIMUM_SLOPE = 0.1
        max_imbalance_fee = channel_capacity * proportional_imbalance_fee / 1e6
    
        assert proportional_imbalance_fee / 1e6 <= MAXIMUM_SLOPE / 2, "Too high imbalance fee"
    
        # calculate function parameters
        s = MAXIMUM_SLOPE
        c = max_imbalance_fee
        o = channel_capacity / 2
        b = s * o / c
        b = min(b, 100)  # limit exponent to keep numerical stability
>       a = c / o ** b
E       OverflowError: (34, 'Result too large')
```

**Comment by palango** [comment]
@kelsos This is the same bug you ran into.

**Comment by palango** [comment]
Can the case `o == 0`  happen?

**Comment by karlb** [comment]
We worked around this by further decreasing the maximum exponent. But we should still verify that
* this does not cause bad curves in some cases
* prevents overflow in all cases

**Comment by Dominik1999** [comment]
closed by #5024
